include/taskolib/SequenceManager.h

```diff
@@ -108,19 +106,18 @@ class SequenceManager
      * is renamed accordingly.
      *
      * \returns a vector containing one SequenceOnDisk object for each sequence that was
-     *          found.
+     *          found. The paths in the returned objects are relative to the root path.
      *
      * \exception Error is thrown if one of the folders needs to be renamed but the
      *            renaming fails.
      */
-    std::vector<SequenceOnDisk> list_sequences() const;
+    std::vector<SequenceOnDisk> list_sequences();
```

Comment:
> `list_sequences()` not being `const` is as bad as DOOCS not-const `get_value()` variant `get()`...

I understand how this looks, but the very concept of constness for this class is dubious. It does not have any data members at all except the base path, which is immutable after construction. And implying that `const` means "does not change anything on disk" is... well, questionable.

Reply:
@alt-graph By using a `const` member function it guarantees read-only because it prohibits any 'internal' member variable modification; they are immutable inside the member function. This applies now and should remain in the future. By removing it we modify the signature. And if there are not any profound reasons of not doing it I would avoid it. I know that we have only one variable (... which is also marked as immutable) but there may be more in the future 😟.

> ..., but the very concept of constness for this class is dubious.

Why do you talk about _class_? It is a member function. Or do I overlook something 🤔. 

@Finii

> The questionable decision to handle old stored sequences 'transparently' (which is not really transparent because they are changed) shows here again.

I understand the point: saving a sequence to the file system should be immutable. By renaming the already stored sequence we modify it. And we have now to distinguish between old stored sequence and the new ones but I don't see any other alternatives to solve it. Moreover I don't have any overview if anybody else already cloned the lib.
Maybe it would be better to run an external script to straighten the sequence folder and remove the dubios distinction in this class - it will declutter the code and make it clearer.

Reply:
> Why do you talk about _class_? It is a member function. Or do I overlook something 🤔.

I just mean that by adding a `const` to a member function we declare that the function may not mutate any members of that *class*. And since that class has (practically) no members, adding `const` has very little practical meaning to what a member function can do. But of course the correct way to handle that is to make all member functions const instead of removing const from them. I have changed the code accordingly.